rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -1,14 +1,51 @@
+require 'activerecord-import'
+
 module RailsEventStoreActiveRecord
   class EventRepository
     def initialize(adapter: Event)
       @adapter = adapter
     end
     attr_reader :adapter
 
-    def create(event, stream_name)
-      data = event.to_h.merge!(stream: stream_name, event_type: event.class)
-      adapter.create(data)
-      event
+    def append_to_stream(events, stream_name, expected_version)
+      events = [*events]
+      expected_version   = case expected_version
+        when nil
+          raise RubyEventStore::InvalidExpectedVersion
+        when :none
+          -1
+        when :auto
+          eis = EventInStream.where(stream: stream_name).order("position DESC").first
+          (eis && eis.position) || -1
+        else
+          expected_version
+      end
+      in_stream = events.flat_map.with_index do |event, index|
+        position = if expected_version == :any
+          nil
+        else
+          expected_version + index + 1
+        end
+        Event.create!(
+          id: event.event_id,
+          data: event.data,
+          metadata: event.metadata,
+          event_type: event.class,
+        )
+        [EventInStream.new(
+          stream:   stream_name,
+          position: position,
+          event_id: event.event_id
+        ),EventInStream.new(
+          stream: "__global__",
+          position: nil,
+          event_id: event.event_id
+        )]
+      end
+      EventInStream.import(in_stream)
+      self
+    rescue ActiveRecord::RecordNotUnique
```

Comment:
this class should know nothing about active record

Reply:
Why would `rails_event_store_active_record/event_repository.rb` need to know nothing about active record when it actively uses active record to implement the needed features.

Reply:
nvm, didn't see we are in active record namespace :)